presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -469,4 +470,25 @@ public String toString()
                 .add("estimatedSize", estimatedSize)
                 .toString();
     }
+
+    public Iterator<Page> getPages(int[] outputChannels)
+    {
+        return new Iterator<Page>() {
+            int currentPosition = 0;
+            PageBuilder pageBuilder = new PageBuilder(types);
+            @Override
+            public boolean hasNext()
+            {
+                pageBuilder.reset();
+                currentPosition = buildPage(currentPosition, outputChannels, pageBuilder);
+                return !pageBuilder.isEmpty();
```

Comment:
Consider this two are very different, I agree we should give them different names. getPgesByIndex sounds fine to me. Or we can just name it buildPages? It's basically an iterator on top of the buildPage API.